### PR TITLE
[subgraph] upcoming feature set for the subgraph

### LINKED ISSUE #1887: [SUBGRAPH] Fix Correctness of GDA Entities
## Why

We currently have correctness issues related to GDA subgraph entities. This is the EPIC to capture all the related issues.

## What

We will fix the correctness in GDA entities, testing with a feature-branch rollout, release a new subgraph package, then roll out too all networks.

### PR DESCRIPTION
**Pull request by kasparkallas**
**What?**
A non-production git branch with all the upcoming features and bug fixes for the subgraph.

**Why?**
Will ease the integration testing with the SDKs and the user-facing products when we deploy the feature branch. This hopefully reduces the count of re-deployments and re-indexes we need to do.

**Other**
Includes:
* https://github.com/superfluid-finance/protocol-monorepo/pull/1877
* https://github.com/superfluid-finance/protocol-monorepo/pull/1878
* https://github.com/superfluid-finance/protocol-monorepo/issues/1887=
* https://github.com/superfluid-finance/protocol-monorepo/pull/1843
* https://github.com/superfluid-finance/protocol-monorepo/pull/1908
* https://github.com/superfluid-finance/protocol-monorepo/pull/1909

### CODE REVIEW

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -1468,20 +1503,39 @@ type ApprovalEvent implements Event @entity(immutable: true) {
...
     """
-    The address that will be granted allowance to transfer the NFT.
+    The address that will be granted allowance to transfer ERC20/NFT.
     """
     to: Account!
+
     """
+    Indicates whether the event was emitted for the approval of an NFT.
+    """
+    isNFTApproval: Boolean!
```
Comment (kasparkallas) [inline packages/subgraph/schema.graphql:1529]: needs to be mapped in the sdk-core, checking

File: packages/subgraph/schema.graphql [hunk 2]
```diff
@@ -1757,6 +1815,11 @@ type Pool @entity {
     totalAmountInstantlyDistributedUntilUpdatedAt: BigInt!
     totalAmountFlowedDistributedUntilUpdatedAt: BigInt!
     totalAmountDistributedUntilUpdatedAt: BigInt!
```
Comment (kasparkallas) [inline packages/subgraph/schema.graphql:1817]: it's preferrable to get this value from the RPC now for a pool member, but it's beneficial to write out how the new fields change the formula on the client-side though

File: packages/subgraph/src/mappings/superfluidPool.ts [hunk 3]
```diff
@@ -48,38 +51,44 @@ export function handleDistributionClaimed(event: DistributionClaimed): void {
...
     const previousUnits = poolMember.units;
     const unitsDelta = event.params.newUnits.minus(previousUnits);
+    const newTotalUnits = pool.totalUnits.plus(unitsDelta);
 
     pool = updatePoolTotalAmountFlowedAndDistributed(event, pool);
+    settlePDPoolMemberMU(pool, poolMember, event.block);
 
-    poolMember = updatePoolMemberTotalAmountUntilUpdatedAtFields(pool, poolMember);
+    // @note TODO update the pool.perUnitFlowRate
+    // @note TODO update the poolMember.perUnitFlowRate
```
Comment (kasparkallas) [inline packages/subgraph/src/mappings/superfluidPool.ts]: some TODO's here

Comment (0xdavinchee) [reply packages/subgraph/src/mappings/superfluidPool.ts]: removed the TODO's, they were addressed

File: packages/subgraph/config/polygon-mainnet.json
Comment (d10r) [inline packages/subgraph/config/polygon-mainnet.json:1]: I think this wasn't added on purpose, or was it?

**Review by d10r — DISMISSED** [review]

File: packages/subgraph/config/polygon-mainnet.json
Comment (kasparkallas) [reply packages/subgraph/config/polygon-mainnet.json:1]: removed the mistakenly added files

**Review by 0xdavinchee — APPROVED** [review]
LGTM

File: packages/subgraph/src/addresses.template.ts [hunk 4]
```diff
@@ -15,3 +15,7 @@ export function getResolverAddress(): Address {
 export function getNativeAssetSuperTokenAddress(): Address {
     return Address.fromString("{{nativeAssetSuperTokenAddress}}");
 }
+
+export function getIsLocalIntegrationTesting(): boolean {
```
Comment (hellwolf) [inline packages/subgraph/src/addresses.template.ts:19]: nitpicking here: `getIs...` is so meh :/ :) . `is...` is idiomatic enough.

**Review by hellwolf — APPROVED** [review]
Quality. Let's go.

**Pull request merged**